src/util/GraphReader.java

```diff
@@ -0,0 +1,62 @@
+package util;
+
+import graph.DirectedGraph;
+import graph.Edge;
+import graph.UndirectedGraph;
+import java.util.LinkedList;
+import java.util.List;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+
+public class GraphReader {
+
+  private String sep;  // edges format "src sep dst"
+
+  public GraphReader(String sep) {
+    this.sep = sep;
+  }
+
+  public String getSep() {
+    return sep;
+  }
+
+  public void setSep(String sep) {
+    this.sep = sep;
+  }
+
+  public DirectedGraph readDirectedGraph(String filename) {
+    DirectedGraph g = new DirectedGraph();
+    for (Edge e : this.readEdges(filename)) {
+      g.addEdge(e);
+    }
+    return g;
+  }
+
+  public UndirectedGraph readUndirectedGraph(String filename) {
+    UndirectedGraph g = new UndirectedGraph();
+    for (Edge e : this.readEdges(filename)) {
+      g.addEdge(e);
+    }
+    return g;
+  }
+
+  private List<Edge> readEdges(String filename) {
+    List<Edge> edges = new LinkedList<>();
+    try (BufferedReader br = new BufferedReader(new FileReader(filename))) {
+      String line;
+      while ((line = br.readLine()) != null) {
+        if (!line.startsWith("#")) {
+          String[] parts = line.split(this.getSep(), 2);
+          int begin = Integer.parseInt(parts[0]);
```

Comment:
What if it can not be split into two parts? Shall consider this. Also, what if each part is not Integer? In the future, you may want to handle the other kinds of input, so I'd suggest making GraphReader general as a base class, and for now make EdgeListReader as an extension of GraphReader.